dio/lib/src/dio.dart

```diff
@@ -43,18 +43,18 @@ import 'entry_stub.dart'
 ///  ```
 
 abstract class Dio {
-  factory Dio([BaseOptions options]) => createDio(options);
+  factory Dio([BaseOptions? options]) => createDio(options);
 
   /// Default Request config. More see [BaseOptions] .
-  BaseOptions options;
+  BaseOptions? options;
```

Comment:
`options` can be non-nullable. I believe there are different ways to achieve that...
If you look around for its usages it is always assigned `BaseOptions()` as a default value, and every access uses `options!.foo`. I believe there is no reason to make it nullable.